src/Foundation/ProtocolAttribute.cs

```diff
@@ -34,6 +34,21 @@ public sealed class ProtocolAttribute : Attribute {
 		public Type WrapperType { get; set; }
 		public string Name { get; set; }
 		public bool IsInformal { get; set; }
+		// In which SDK version this protocol switched from (or to, depending on the current IsInformal value) being informal.
+		// If IsInformal = true, then this protocol switched to being informal in the specified SDK version
+		// If IsInformal = false, then this protocol switched to a real protocol in the specified SDK version
+		// Version is not a valid type for attributes, so we're using an array of ints.
+		int[] informal_switch;
+		public int [] InformalSwitch {
```

Comment:
I didn't consider using a string, but I don't think it's much of a difference cost-wise (and in any case this is rarely needed).

I checked whether the metadata is bigger for strings or int arrays, and couldn't find any difference at all up to 100.000 attributes.

So I guess it comes down to readability, and I think strings are slightly more readable, so I'll change it.